bootstrap build broken

see: https://builder.sourceware.org/buildbot/#/builders/179/builds/1242


```
 /home/runner/work/gccrs/gccrs/gccrs-build/./prev-gcc/xg++ -B/home/runner/work/gccrs/gccrs/gccrs-build/./prev-gcc/ -B/usr/local/x86_64-pc-linux-gnu/bin/ -nostdinc++ -B/home/runner/work/gccrs/gccrs/gccrs-build/prev-x86_64-pc-linux-gnu/libstdc++-v3/src/.libs -B/home/runner/work/gccrs/gccrs/gccrs-build/prev-x86_64-pc-linux-gnu/libstdc++-v3/libsupc++/.libs  -I/home/runner/work/gccrs/gccrs/gccrs-build/prev-x86_64-pc-linux-gnu/libstdc++-v3/include/x86_64-pc-linux-gnu  -I/home/runner/work/gccrs/gccrs/gccrs-build/prev-x86_64-pc-linux-gnu/libstdc++-v3/include  -I/home/runner/work/gccrs/gccrs/libstdc++-v3/libsupc++ -L/home/runner/work/gccrs/gccrs/gccrs-build/prev-x86_64-pc-linux-gnu/libstdc++-v3/src/.libs -L/home/runner/work/gccrs/gccrs/gccrs-build/prev-x86_64-pc-linux-gnu/libstdc++-v3/libsupc++/.libs  -fno-PIE -c  -DIN_GCC_FRONTEND -g -O2 -fno-checking -gtoggle -DIN_GCC     -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wmissing-format-attribute -Wconditionally-supported -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings -Werror -fno-common  -DHAVE_CONFIG_H -I. -I. -I../../gcc -I../../gcc/. -I../../gcc/../include  -I../../gcc/../libcpp/include -I../../gcc/../libcody  -I../../gcc/../libdecnumber -I../../gcc/../libdecnumber/bid -I../libdecnumber -I../../gcc/../libbacktrace   -o i386-rust.o -MT i386-rust.o -MMD -MP -MF ./.deps/i386-rust.TPo ../../gcc/config/i386/i386-rust.cc
../../gcc/rust/util/rust-unicode.cc:50:51: error: ‘uint32_t’ is not a member of ‘std’; did you mean ‘wint_t’?
   50 | binary_search_sorted_array (const std::array<std::uint32_t, SIZE> &array,
      |                                                   ^~~~~~~~
      |                                                   wint_t
../../gcc/rust/util/rust-unicode.cc:50:65: error: template argument 1 is invalid
   50 | binary_search_sorted_array (const std::array<std::uint32_t, SIZE> &array,
      |                                                                 ^
../../gcc/rust/util/rust-unicode.cc: In function ‘int64_t Rust::binary_search_sorted_array(const int&, uint32_t)’:
../../gcc/rust/util/rust-unicode.cc:63:16: error: invalid types ‘const int[uint32_t {aka unsigned int}]’ for array subscript
   63 |       if (array[mid] <= target)
      |                ^
../../gcc/rust/util/rust-unicode.cc:70:12: error: invalid types ‘const int[uint32_t {aka unsigned int}]’ for array subscript
   70 |   if (array[mid] == target)
      |            ^
../../gcc/rust/util/rust-unicode.cc: In function ‘bool Rust::is_numeric(uint32_t)’:
../../gcc/rust/util/rust-unicode.cc:218:44: error: no matching function for call to ‘binary_search_sorted_array(const std::array<unsigned int, 1831>&, uint32_t&)’
  218 |   int64_t res = binary_search_sorted_array (NUMERIC_CODEPOINTS, codepoint);
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../gcc/rust/util/rust-unicode.cc:50:1: note: candidate: ‘template<long unsigned int SIZE> int64_t Rust::binary_search_sorted_array(const int&, uint32_t)’
   50 | binary_search_sorted_array (const std::array<std::uint32_t, SIZE> &array,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
../../gcc/rust/util/rust-unicode.cc:50:1: note:   template argument deduction/substitution failed:
../../gcc/rust/util/rust-unicode.cc:218:44: note:   couldn’t deduce template parameter ‘SIZE’
  218 |   int64_t res = binary_search_sorted_array (NUMERIC_CODEPOINTS, codepoint);
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[3]: *** [../../gcc/rust/Make-lang.in:407: rust/rust-unicode.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```

--- a/gcc/rust/util/rust-unicode.cc
+++ b/gcc/rust/util/rust-unicode.cc
@@ -47,7 +47,7 @@ binary_search_ranges (
 
 template <std::size_t SIZE>
 int64_t
-binary_search_sorted_array (const std::array<std::uint32_t, SIZE> &array,
+binary_search_sorted_array (const std::array<uint32_t, SIZE> &array,
 			    uint32_t target)
 {
   // FIXME: use binray search function from <algorithm>

